semgrep-core/src/parsing/tree_sitter/Parse_swift_tree_sitter.ml

```diff
@@ -47,6 +47,18 @@ let in_pattern env =
   | Program -> false
   | Pattern -> true
 
+let entity_of_pattern (pat : G.pattern) : G.entity =
```

Comment:
We might want to move this to AST_generic_helpers.ml

Reply:
Ok. I think it could be useful for OCaml, Javascript where the left hand part of a definition can be a complex pattern.
Anyway, let's keep it like that and next time I see the need in another language I'll factorize.